src/main/drivers/bus.h

```diff
@@ -54,20 +76,25 @@ typedef struct busDeviceDescriptor_s {
             uint8_t     address;
         } i2c;
     } busdev;
+    ioTag_t irqPin;
 } busDeviceDescriptor_t;
 
 typedef struct busDevice_s {
-    busType_e           busType;
+    const busDeviceDescriptor_t * descriptor;
+    busType_e busType;              // Bus type 
     union {
         struct {
-            SPIDevice   spiBus;
-            IO_t        csnPin;
+            SPIDevice spiBus;       // SPI bus ID
+            IO_t csnPin;            // IO for CS# pin
         } spi;
         struct {
-            I2CDevice   i2cBus;
-            uint8_t     address;
+            I2CDevice i2cBus;       // I2C bus ID
+            uint8_t address;        // I2C bus device address
         } i2c;
     } busdev;
+    IO_t irqPin;                    // Device IRQ pin. Bus system will only assign IO_t object to this var. Initialization is up to device driver
```

Comment:
I think the device driver should probably own the `irqPin`.

Reply:
It's an abstraction of a generic device on a bus. Most devices have an IRQ pin. We might not use it, but we have it. Allocating and handling it in one place is more logical than to allow each of the drivers handle it on it's own.

I'm even thinking about handling incoming interrupts in the `busDevice_t` layer.

Reply:
But it's not handled in one place. Your comment specifically says `Bus system will only assign IO_t object to this var. Initialization is up to device driver`

Reply:
@martinbudden that's temporary. The code is far from completion. Eventually IRQ handling code will be migrated to `busDevice_t` and only callbacks & flags will be exposed to device driver.